### PR TITLE
feat:refactor: kind aware cmd add

### PR DESCRIPTION
**Pull request by acpana**
This work comes back to look at the controller builder cmd.

* it reverts the per package work as we're moving away from that direction to having a structure like:
```
resourceA_controller.go
resourceB_controller.go
```
* For any datastructure or function that needs a resource specific name, I added the necessary changes in the template.

Been testing this with the `BigQueryAnalyticsHubDataExchange` and `BigQueryAnalyticsHubListing`

### CODE REVIEW

File: dev/tools/controllerbuilder/template/controller/controller.go [hunk 1]
```diff
@@ -78,20 +78,20 @@ const (
...
 
-var _ directbase.Model = &model{}
+var _ directbase.Model = &model{{.Kind}}{}
 
-type model struct {
+type model{{.Kind}} struct {
 	config config.ControllerConfig
 }
 
-func (m *model) client(ctx context.Context) (*gcp.Client, error) {
```
Comment (jingyih) [inline dev/tools/controllerbuilder/template/controller/controller.go:94]: nit: `clientForKind` seems a bit verbose? Given that the function already has a pointer receiver of `modelKind`.

**Review by jingyih — COMMENTED** [review]
/lgtm

**Review by jingyih — APPROVED** [review]
/lgtm
/approve

/hold
Feel free to remove when PR is ready to merge.

**Comment by yuwenma** [comment]
Is this for using the mapping?

**Pull request merged**